nav2_bringup/launch/tb3_gz_simulation_launch.py

```diff
@@ -0,0 +1,326 @@
+        condition=IfCondition(use_simulator),
+        cmd=['ros2', 'control', 'load_controller', '--set-state', 'active',
+             'joint_state_broadcaster'],
+        output='screen'
+    )
+    load_diffdrive_controller = ExecuteProcess(
+        condition=IfCondition(use_simulator),
+        cmd=['ros2', 'control', 'load_controller', '--set-state', 'active',
+             'diffdrive_controller'],
+        output='screen'
+    )
+    spawn_model = Node(
+        condition=IfCondition(use_simulator),
+        package='ros_gz_sim',
+        executable='create',
+        output='screen',
+        arguments=[
+            '-entity', robot_name,
+            '-string', Command(['xacro', ' ', robot_sdf]),
+            '-robot_namespace', namespace,
+            '-x', pose['x'], '-y', pose['y'], '-z', pose['z'],
+            '-R', pose['R'], '-P', pose['P'], '-Y', pose['Y']]
+    )
+    gazebo_server = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution([
+                FindPackageShare('ros_gz_sim'),
+                'launch',
+                'gz_sim.launch.py'
+            ])
+        ),
+        launch_arguments={'gz_args': ['-r -s ', world]}.items(),
+        condition=IfCondition(use_simulator)
+    )
+
+    gazebo_client = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution([
+                FindPackageShare('ros_gz_sim'),
+                'launch',
+                'gz_sim.launch.py'
+            ])
+        ),
+        condition=IfCondition(PythonExpression(
+            [use_simulator, ' and not ', headless])),
+        launch_arguments={'gz_args': ['-g ']}.items(),
+    )
+
+    # Create the launch description and populate
+    ld = LaunchDescription()
+
+    # Declare the launch options
+    ld.add_action(declare_namespace_cmd)
+    ld.add_action(declare_use_namespace_cmd)
+    ld.add_action(declare_slam_cmd)
+    ld.add_action(declare_map_yaml_cmd)
+    ld.add_action(declare_use_sim_time_cmd)
+    ld.add_action(declare_params_file_cmd)
+    ld.add_action(declare_autostart_cmd)
+    ld.add_action(declare_use_composition_cmd)
+
+    ld.add_action(declare_rviz_config_file_cmd)
+    ld.add_action(declare_use_simulator_cmd)
+    ld.add_action(declare_use_robot_state_pub_cmd)
+    ld.add_action(declare_use_rviz_cmd)
+    ld.add_action(declare_simulator_cmd)
+    ld.add_action(declare_world_cmd)
+    ld.add_action(declare_robot_name_cmd)
+    ld.add_action(declare_robot_sdf_cmd)
+    ld.add_action(declare_use_respawn_cmd)
+
+    ld.add_action(start_robot_state_publisher_cmd)
+
+    ld.add_action(set_env_vars_resources)
+    ld.add_action(gazebo_server)
+    ld.add_action(gazebo_client)
+
+    # Add any conditioned actions
+    ld.add_action(clock_bridge)
+    ld.add_action(lidar_bridge)
+    ld.add_action(imu_bridge)
+    ld.add_action(spawn_model)
+    ld.add_action(load_joint_state_broadcaster)
+    ld.add_action(load_diffdrive_controller)
```

Comment:
Might be value in having a sub-level nav2_ign_sim.launch.py that gets called at this level with maybe an sim: [ign, gz] param to enable easier modifications for a nav2_gz_sim.launch.py? (includes all bridge/ros_gz stuff as well) 
@SteveMacenski maybe add that as a stretch goal?

Reply:
I don't understand this comment, sorry?

Reply:
I could have probably written that more clearly. I guess what I'm trying to describe is passing launch args to a lower level nav2_ign_sim or nav2_gz_sim launch scripts that can include all the ros_ign or ros_gz spawn and bridge commands. So they can be more generalized and used for other vehicles as well.

Reply:
@ahcorde what do you think about that? I think that could help generalize the simulation so folks can swap out with their robots by changing 1 launch file with all the robot-gazebo specifics in one place. 

Particularly, all the different bridges needed so that there's a clear spot to add more sensors. I suppose adding the SDF stuff too would make sense to make all the robot-specifics in one place + have the sensor bridges next to the SDF asking for them.

Reply:
sure, I can break the launch file in some peaces, `gz_bridges.launch.py` and `gz_nav.launch.py` sounds good ?

Reply:
Leave this file named as it is since its consistent with the other launch file's names. I'd just add a new ones like `gz_robot_launch.py` or similar for the gz-robot specifics

Reply:
https://github.com/ros-planning/navigation2/pull/3634/commits/792569a9518eee7a4fcf94f58e4b4493bd49a0e4